openedx/core/djangoapps/user_authn/cookies.py

```diff
@@ -77,80 +124,73 @@ def set_logged_in_cookies(request, response, user):
-    # will transmit it only over secure connections.
-    #
-    # In non-production environments (acceptance tests, devstack, and sandboxes),
-    # we still want to set this cookie.  However, we do NOT want to set it to "secure"
-    # because the browser won't send it back to us.  This can cause an infinite redirect
-    # loop in the third-party auth flow, which calls `is_logged_in_cookie_set` to determine
-    # whether it needs to set the cookie or continue to the next pipeline stage.
-    user_info_cookie_is_secure = request.is_secure()
-    user_info = get_user_info_cookie_data(request)
+    The user info cookie has the following format:
+    {
+        "version": 1,
+        "username": "test-user",
+        "header_urls": {
+            "account_settings": "https://example.com/account/settings",
+            "resume_block":
+                "https://example.com//courses/org.0/course_0/Run_0/jump_to/i4x://org.0/course_0/vertical/vertical_4"
+            "learner_profile": "https://example.com/u/test-user",
+            "logout": "https://example.com/logout"
+        }
+    }
+    """
+    cookie_settings = standard_cookie_settings(request)
 
+    user_info = _get_user_info_cookie_data(request, user)
     response.set_cookie(
         settings.EDXMKTG_USER_INFO_COOKIE_NAME.encode('utf-8'),
         json.dumps(user_info),
-        secure=user_info_cookie_is_secure,
         **cookie_settings
     )
 
 
-def set_experiments_is_enterprise_cookie(request, response, experiments_is_enterprise):
-    """ Sets the experiments_is_enterprise cookie on the response.
-    This cookie can be used for tests or minor features,
-    but should not be used for payment related or other critical work
-    since users can edit their cookies
-    """
-    cookie_settings = standard_cookie_settings(request)
-    # In production, TLS should be enabled so that this cookie is encrypted
-    # when we send it.  We also need to set "secure" to True so that the browser
-    # will transmit it only over secure connections.
-    #
-    # In non-production environments (acceptance tests, devstack, and sandboxes),
-    # we still want to set this cookie.  However, we do NOT want to set it to "secure"
-    # because the browser won't send it back to us.  This can cause an infinite redirect
-    # loop in the third-party auth flow, which calls `is_logged_in_cookie_set` to determine
-    # whether it needs to set the cookie or continue to the next pipeline stage.
-    cookie_is_secure = request.is_secure()
-
-    response.set_cookie(
-        'experiments_is_enterprise',
-        json.dumps(experiments_is_enterprise),
-        secure=cookie_is_secure,
-        **cookie_settings
-    )
+def _set_jwt_cookies(response, request, user):  # pylint: disable=unused-argument
+    """ Sets a cookie containing a JWT on the response. """
+    if not JWT_COOKIES_FLAG.is_enabled():
+        return
+    # TODO
```

Comment:
Remind me to return to this when it is ready, or do you plan on landing this like this?  If so, can you add the JIRA ticket for completing this work?

Reply:
Yes, I plan to land it this way.  I added a reference to ARCH-236.